### DISCUSSION TITLE
可否在add_package的时机再增加一个on_xxx的回调

### DISCUSSION THREAD

**Discussion by augustheart** [xmake-io/xmake#3038] [Ideas]
也是来自上一个帖子倒腾wxwidgets的想法。wx的正规用法是需要使用wx-config获取相关的-I和-L参数。但是在xmake的机制下，wx-config位于wx包的bin目录下面，在build的时候，很难进入到这个包获取，而且似乎也需要在最终的targe下面获取。
能不能提供一个时机，当包被引入target的时候，动态获取需要的参数，这样相关的配置可以写在package中，最终的target不需要关心。我开始以为可以在on_load的时候获取，但是测试下来on_load并不符合要求。

当然，这也可能只是我用得不够熟练，没有发现方法。找了一下repo里面，也没找到灵感。

**Comment by waruqi** [comment]
用了 xmake 包，就没必要用 wx-config 了，相关 includedirs linkdirs 已经被安装到 对应包目录下，会自动获取整合，别去管 wx-config，只需要 add_packages

**Reply by waruqi** [reply]
参考：https://github.com/xmake-examples/wxWidgets-scaffold
